howtos/module/_posts/2016-06-30-Ten-ways-improve-modules.md

```diff
@@ -0,0 +1,110 @@
+---
+layout: post
+title:  "Ten ways to improve your PrestaShop modules!"
+subtitle: "Stop the bugs, get back to innovating!"
+date: 2016-07-01 11:07:11
+author: [ mdekker ]
+icon: icon-puzzle
+tags: [ howto, tutorial, modules ]
+published: false
+---
+
+PrestaShop has numerous features in the API, both hidden in its code as well as in the developer documentation, that allow you to create your own module in no time!
+Unfortunately I have noticed that many developers do not use these features to their advantage and I think it's time to change all that :)
+Hopefully, these tips will increase your development speed and save you lots and lots of time. And if not satisfied, you'll get your money back! (Oh, wait...)  
+
+## The time savers
+Here is a list of very very useful tools and features (if you ask me). Prepare for some technical stuff as we're going to cover some of them in depth.
+
+### 1. Use an Integrated Development Environment
+Using an integrated development environment such as PhpStorm or NetBeans is a great way to improve your productivity.
+Tools such as these come with dozens of nifty features out-of-the-box to develop quicker... a lot quicker!
+
+As the name implies, they intend to integrate as many features as possible, allowing you for example to automatically upload your changes over FTP. It is also possible make a direct connection with the MySQL database and browse it from your IDE (not recommended for production environments!) to see what was actually written to the database by your module.  
+An IDE is aware of your code, guiding you while you write your code, wherever and whenever you need it. Cool tip for PhpStorm users: navigate to *Help > Productivity guide*. It will show you how many keystrokes it has saved you, as well as amount of bugs prevented and hours saved. The numbers will add up quickly after using it for a while!
+These IDEs were designed to never take your hands off the keyboard while performing basic programming tasks. Feel free to configure the keyboard shortcuts in a way you're comfortable with and don't disable the "Tip of the day" messages. They can make the "steep" learning curve of using an IDE a little bit less steep.
+
+### 2. Stick to coding standards
+PrestaShop has changed their coding standards several times in the past and as of this month, another coding standard has gone into effect. Why does PrestaShop change its [coding standards]({% post_url 2016-05-10-prestashop-coding-standards %})?  
```

Comment:
Well, we had two: PS's own standard, then for v1.6.1.0 (released July 2015) we adopted PSR-2, the latest evolution still uses PSR-2 with some Symfony additions, and details the coding standards for other languages (HTML, CSS, JavaScript), because there was none before.
Also, "as of this month" shows the age of the article: the evolution of the coding standards was put in place on May 10th :)